CMakeLists.txt

```diff
@@ -4,6 +4,30 @@
 
 cmake_minimum_required(VERSION 3.16)
 
+## Some workarounds for platform build quirks
+if(WIN32)
+    ## CMake v3.20 has problems with race conditions in dependency generation.
+    ## See: https://gitlab.kitware.com/cmake/cmake/-/issues/22014
+    if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_EQUAL 3.20)
+        cmake_policy(SET CMP0116 OLD)
+    endif()
+
+    ## CMake also has trouble finding OpenSSL libraries on Windows, and may
+    ## need some help.
+    if(EXISTS "C:/MozillaVPNBuild/SSL" AND NOT DEFINED OPENSSL_ROOT_DIR)
```

Comment:
For context, we need to explicitly set `OPENSSL_ROOT_DIR` and call `find_package(OpenSSL REQUIRED)` whenever we are using the SSL libs from [qt_static_windows](https://github.com/mozilla-mobile/qt_static_windows) repository, or we wind up with failures to find the Qt packages. This also means that developers would also need to remember to set `OPENSSL_ROOT_DIR` whenever they're trying to build locally.

It isn't necessary when using a dynamically-linked build of Qt from the Qt online installer, so I figured it would be cleaner to make this automated instead of trying to explain the nuance of when and when not to set `OPENSSL_ROOT_DIR`

But yes, we could remove this hack and make it an argument in CI.